Sources/Tracking/CustomerIOImplementation.swift

```diff
@@ -229,6 +233,10 @@ internal class CustomerIOImplementation: CustomerIOInstance {
             logger.info("no profile identified, so not registering device token to a profile")
             return
         }
+        if identifier.isBlankOrEmpty() {
+            logger.error("profile cannot be identified: Identifier is empty, so not registering device token to a profile")
+            return
+        }
```

Comment:
```suggestion
```

The identifier should never be blank or empty because the only place that `profileStore.identifier` gets populated is in the `identify()` function. So this `if` statement will never be `true`. 

Also, I don't think that it's the responsibility of `registerDeviceToken` to assert that a profile identifier is not empty. If we add this `if` check here, we might as well add that check everywhere that a profile identifier is attached to the HTTP request `CustomerIO.track()`, `CustomerIO.screen()`, etc. 

Therefore, I suggest we remove this.

Reply:
We already have users, who could have identified with a `blank` identifier. But I do think we should always check for this, as this is a breaking app behavior. Even though our only source right now is through `identify` but that `repository` to store identifiers is still exposed to other classes and by mistake or anything that can still be updated. 

So i think, this checks adds a safe-guard.

Reply:
Users can also register device token using public method `registerDeviceToken`. They may not provide a nil value as this method needs a non-nil value but sending `"   "` is not an exception. The chances of a user sending such values are grim but not to be ignored. I think having this check is a good approach here.

Reply:
> We already have users, who could have identified with a blank identifier. 

Yes, that is true that scenario could exist. To help get this fix out, we could make a new PR to support the scenario of fixing SDK installs today where the identifier is blank. 

I suggest a future PR because (1) I do believe that use case is beyond the scope of this PR and (2) I would rather use a different solution to handle that use case then what is currently implemented in this PR.

Reply:
I think we should let go this PR and remove this code snippet when we make another PR. Same check has been done on Android as well and it is now live. This is an OpsBug issue and need to be closed as soon as possible when the fix is done.